src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs

```diff
@@ -1,150 +1,143 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the MIT License. See LICENSE in the project root for license information.
-
-using System;
-using DurableTask.AzureStorage;
-using Microsoft.Extensions.Options;
-using Newtonsoft.Json;
-
-namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
-{
-    internal class AzureStorageDurabilityProviderFactory : IDurabilityProviderFactory
-    {
-        private readonly DurableTaskOptions options;
-        private readonly AzureStorageOptions azureStorageOptions;
-        private readonly IConnectionStringResolver connectionStringResolver;
-        private readonly AzureStorageOrchestrationServiceSettings defaultSettings;
-        private readonly string defaultConnectionName;
-        private AzureStorageDurabilityProvider defaultStorageProvider;
-
-        public AzureStorageDurabilityProviderFactory(
-            IOptions<DurableTaskOptions> options,
-            IConnectionStringResolver connectionStringResolver)
-        {
-            this.options = options.Value;
-            this.azureStorageOptions = new AzureStorageOptions();
-            JsonConvert.PopulateObject(JsonConvert.SerializeObject(this.options.StorageProvider), this.azureStorageOptions);
-
-            this.azureStorageOptions.Validate();
-            if (!this.options.IsDefaultHubName())
-            {
-                this.azureStorageOptions.ValidateHubName(this.options.HubName);
-            }
-            else if (!this.azureStorageOptions.IsSanitizedHubName(this.options.HubName, out string sanitizedHubName))
-            {
-                this.options.SetDefaultHubName(sanitizedHubName);
-            }
-
-            this.connectionStringResolver = connectionStringResolver;
-            this.defaultConnectionName = this.azureStorageOptions.ConnectionStringName ?? ConnectionStringNames.Storage;
-            this.defaultSettings = this.GetAzureStorageOrchestrationServiceSettings();
-        }
-
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using DurableTask.AzureStorage;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal class AzureStorageDurabilityProviderFactory : IDurabilityProviderFactory
+    {
+        private readonly DurableTaskOptions options;
+        private readonly AzureStorageOptions azureStorageOptions;
+        private readonly IConnectionStringResolver connectionStringResolver;
+        private readonly string defaultConnectionName;
+        private readonly Lazy<AzureStorageOrchestrationServiceSettings> defaultSettings;

```

Comment:
I think the approach with `Lazy` is actually just a more graceful version of what I tried to do in my PR. I might change my approach to utilize this.